Prototope/Image.swift

```diff
@@ -34,4 +34,127 @@ public struct Image {
+        height = Int(image.size.height)
+        
+        data = UnsafeMutablePointer<Pixel>.alloc(width*height)
+        dataDestroyer = PixelDataDestroyer(data: data, width: width, height: height)
+        
+        let colorSpace: CGColorSpace = CGColorSpaceCreateDeviceRGB()
+        let bitmapInfo = CGBitmapInfo(CGImageAlphaInfo.PremultipliedLast.rawValue)
+        let bytesPerRow = width * 4
+        
+        let context = CGBitmapContextCreate(data, width, height, 8, bytesPerRow, colorSpace, bitmapInfo)
+        
+        CGContextDrawImage(context, CGRect(origin: CGPointZero, size: image.uiImage.size), image.uiImage.CGImage)
+    }
+    
+    public subscript (position: Int) -> Pixel {
+        get { return data[position] }
+        set { data[position] = newValue }
+    }
+    
+    public func pixelAt(row: Int, column: Int) -> Pixel {
+        let idx = row * width + column
+        return self[idx]
+    }
+    
+    public func generate() -> IndexingGenerator<PixelBitmap> {
+        return IndexingGenerator(self)
+    }
+    
+    public func toImage() -> Image {
+        let colorSpace: CGColorSpace = CGColorSpaceCreateDeviceRGB()
+        let bitmapInfo = CGBitmapInfo(CGImageAlphaInfo.PremultipliedLast.rawValue)
+        let context = CGBitmapContextCreate(data, width, height, 8, width * 4, colorSpace, bitmapInfo)
+        let cgImage = CGBitmapContextCreateImage(context)
+        
+        return Image(UIImage(CGImage: cgImage)!)
+    }
+    
+    public func transform(transform: Pixel -> Pixel) -> PixelBitmap {
+        var newBitmap = self
+        for idx in startIndex..<endIndex {
+            newBitmap[idx] = transform(self[idx])
+        }
+        
+        return newBitmap
+    }
+    
+    public func transform(transform: (position: Int, Pixel) -> Pixel) -> PixelBitmap {
+        var newBitmap = self
+        for idx in startIndex..<endIndex {
+            newBitmap[idx] = transform(position: idx, self[idx])
+        }
+        
+        return newBitmap
+    }
+    
+    public func transform(transform: (row: Int, column: Int, Pixel) -> Pixel) -> PixelBitmap {
+        var newBitmap = self
+        for idx in startIndex..<endIndex {
+            let row = idx % width
+            let column = idx - (row * width)
+            newBitmap[idx] = transform(row: row, column:column, self[idx])
+        }
+        
+        return newBitmap
+    }
+    
+    private class PixelDataDestroyer {
+        let pixelDataToCleanUp: UnsafeMutablePointer<Pixel>
+        let width: Int
+        let height: Int
+        
+        init(data: UnsafeMutablePointer<Pixel>, width: Int, height: Int) {
+            self.pixelDataToCleanUp = data
+            self.width = width
+            self.height = height
+        }
+        
+        deinit {
+            pixelDataToCleanUp.dealloc(width*height)
+        }
+    }
+}
+
+
+/** A representation of a single pixel in an RGBA bitmap image. */
+public struct Pixel {
```

Comment:
To hide the `UInt8` ugliness, you could name these properties `redRaw` and expose `red:  Float` as a computed property.